app/main.py

```diff
@@ -26,81 +26,65 @@
 
-import config as cfg  # Cf. config.py file
 import dash
 import dash_bootstrap_components as dbc
-
-# Various modules provided by Dash and Dash Leaflet to build the page layout
 import dash_core_components as dcc
 import dash_html_components as html
-import dash_leaflet as dl
-
-# Various utils
 import numpy as np
-
-# Pandas to read the login correspondences file
 import pandas as pd
-
-# Import used to make the API call in the login callback
 import requests
+import sentry_sdk
+from dash.dependencies import ALL, MATCH, Input, Output, State
+from dash.exceptions import PreventUpdate
+from flask_caching import Cache
+from pyroclient import Client
+from sentry_sdk.integrations.flask import FlaskIntegration
 
-# From alert_screen.py, we import the main layout instantiation function and some others needed for interactivity
-from alert_screen import (
+import config as cfg
+from pages.device_status import DashboardScreen, build_dashboard_table
+from pages.homepage import Homepage, choose_map_style
+from pages.screen import (
     AlertScreen,
     build_alert_detected_screen,
     build_no_alert_detected_screen,
 )
-from alerts import (
+from services import api_client
+from utils._utils import choose_layer_style, is_hour_between
+from utils.alerts import (
     build_alert_overview,
     build_alerts_elements,
     build_individual_alert_components,
     build_sites_markers,
     display_alert_selection_area,
     retrieve_site_from_device_id,
 )
-from dash.dependencies import ALL, MATCH, Input, Output, State
-from dash.exceptions import PreventUpdate
-
-# From dashboard_screen.py, we import the main layout instantiation function
-from dashboard_screen import DashboardScreen, build_dashboard_table
-
-# Flask caching import
-from flask_caching import Cache
-
-# From other Python files, we import some functions needed for interactivity
-# From homepage.py, we import the main layout instantiation function
-from homepage import Homepage, choose_map_style
-
-# From the pyroclient package
-from pyroclient import Client
-from risks import build_risks_geojson_and_colorbar
-
-# Importing the pre-instantiated Pyro-API client
-from services import api_client
-from utils import (
-    build_filters_object,
-    build_legend_box,
-    choose_layer_style,
-    is_hour_between,
-)
 
 # --- Imports from other Python files
 
 
 # ----------------------------------------------------------------------------------------------------------------------
 # APP INSTANTIATION & OVERALL LAYOUT
+logger = logging.getLogger("uvicorn.error")
+# Sentry
+if isinstance(cfg.SENTRY_DSN, str):
+    sentry_sdk.init(
+        dsn=cfg.SENTRY_DSN,
+        release=cfg.VERSION,
+        server_name=cfg.SERVER_NAME,
+        environment="production" if isinstance(cfg.SERVER_NAME, str) else None,
+        integrations=[
+            FlaskIntegration(),
+        ],
+        traces_sample_rate=1.0,
+    )
+    logger.info(f"Sentry middleware enabled on server {cfg.SERVER_NAME}")
```

Comment:
Could you explain how does sentry interacts with the platform ?

Reply:
It's experimental, but on the platform side, whenever there is a crash, it will send a sentry alert (similar to the API). However it's a bit obscure how to do it with dash, so this is experimental :+1: